hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+
+        outcome = np.digitize(outcome, bins=np.array([np.mean(outcome) - 2])) if self.is_binary_target else outcome
+
+        if self.is_treatment:
+            return np.column_stack((m_with_dummy, treatment, outcome))
+        return np.column_stack((m_with_dummy, outcome))
+
+    def _set_df(self, data: np.ndarray, info: np.ndarray):
+        """
+        Generate names for columns and combine all data to DataFrame
+        Args:
+            data:
+                Array with main features
+            info:
+                Array with info features
+
+        Returns:
+            df:
+                DataFrame with generated data
+        """
+        gender = [["male", "female"][i] for i in np.random.choice(a=np.array([0, 1]), size=self.num_records)]
+        product = [['Deposit', 'Credit', 'Investment'][i] for i in np.random.choice(a=np.array([0, 1, 2]),
+                                                                                    size=self.num_records)]
+        self.treatment_name = ['treatment'] if self.is_treatment else []
+        df = pd.DataFrame(data=data, columns=self.features_names + self.treatment_name + self.target_names)
+        self.treatment_name = 'treatment' if self.is_treatment else []
+        for i in range(self.num_info_cols):
+            df.insert(i, self.info_col_names[i], pd.Series(info[i]))
+        df.insert(self.num_info_cols, 'feature_1', pd.Series(gender))
+        df.insert(self.num_info_cols + 1, 'feature_2', pd.Series(product))
+        self.features_names = ['feature_1', 'feature_2'] + self.features_names
+        return df
+
+    def _set_column_names(self):
+        """
+        Set names for columns in DataFrame
+        """
+        self.features_names = ['feature_' + str(i) for i in range(3, self.num_features + 3)]
+        self.treatment_name = ['treatment'] if self.is_treatment else []
+        self.target_names = ['target_' + str(i) for i in range(1, self.num_targets + 1)] if self.num_targets > 1 else [
+            'target']
+        self.info_col_names = ['info'] if self.num_info_cols == 1 else ['info_' + str(i) for i in
+                                                                        range(1, self.num_info_cols + 1)]
+
+    def _set_nans(self):
+        """
+        Set NaN values to DataFrame according to their respective
+        """
+        if self.na_columns or self.na_step:
+            self.na_step = [self.na_step] if self.na_step else [10]
+            self.na_columns = self.na_columns if self.na_columns else self.features_names + self.info_col_names
+            self.na_columns = [self.na_columns] if isinstance(self.na_columns, str) else self.na_columns
+            if not set(self.na_columns).issubset(self.df.columns):
+                raise KeyError(f'There is no columns {self.na_columns} in data. Only {list(self.df.columns)} provided.')
+            self.na_step = self.na_step[:len(self.na_columns)] if len(self.na_step) > len(
```

Comment:
Ternary expressions are too complicated to understand. Regular if\else blocks are better